internal/fs/dir_handle.go

```diff
@@ -56,36 +55,63 @@ type dirHandle struct {
 	//
 	// GUARDED_BY(Mu)
 	entriesValid bool
+
+	//condition variable is for signalling whether a fresh set of entries has been fetched
+	cond *sync.Cond
+	//error during the fetching goroutine must be communicated using this to the main go routine
```

Comment:
always start comments with capital letter and end with full stop. Add a space between // and first work